### DISCUSSION TITLE
Have event as a json string?

### DISCUSSION THREAD

**Discussion by daledude** [vectordotdev/vector#11058] [Q&A] [answered]
Is there a method where I can take a journald event and encode into a string formatted as json so I can save the whole event into a single clickhouse string field?

**Comment by spencergilbert** [comment, marked as answer]
Hi @daledude I think the [`encode_json`](https://vector.dev/docs/reference/vrl/functions/#encode_json) function in VRL should help you here.

**Reply by daledude** [reply]
Hi @spencergilbert 

Thank you very much for your time. Works perfectly!
```
.payload = encode_json(.)
```
